app/scripts/components/table/list-view.js

```diff
@@ -21,63 +26,61 @@ class List extends React.Component {
     this.updateSelection = this.updateSelection.bind(this);
     this.onBulkActionSuccess = this.onBulkActionSuccess.bind(this);
     this.onBulkActionError = this.onBulkActionError.bind(this);
-    this.config = this.config.bind(this);
+    this.getQueryConfig = this.getQueryConfig.bind(this);
     this.renderSelectAll = this.renderSelectAll.bind(this);
+
+    const initialPage = 1;
+    const initialSortIdx = props.sortIdx || 0;
+    const initialOrder = 'desc';
+
     this.state = {
-      page: 1,
-      sortIdx: props.sortIdx || 0,
-      order: 'desc',
+      page: initialPage,
+      sortIdx: initialSortIdx,
+      order: initialOrder,
       selected: [],
       prefix: null,
-      queryConfig: {},
+      queryConfig: {
+        page: initialPage,
+        order: initialOrder,
+        sort_by: this.getSortProp(initialSortIdx),
+        params: {},
+        ...(props.query || {})
+      },
       params: {},
       completedBulkActions: 0,
       bulkActionError: null
     };
   }
 
-  componentDidMount () {
-    this.setState({ queryConfig: this.config() }); // eslint-disable-line react/no-did-mount-set-state
-  }
-
-  componentDidUpdate (prevProps) {
+  componentDidUpdate (prevProps, prevState) {
     const { query, list, sortIdx } = this.props;
-    if (JSON.stringify(query) !== JSON.stringify(prevProps.query)) {
-      this.setState({ queryConfig: this.config({}, query) }); // eslint-disable-line react/no-did-update-set-state
-    }
-
-    // remove null and undefined values
-    const { params } = list;
-    const validParams = {};
-    for (let key in params) {
-      let value = params[key];
-      if (!undef(value) && value !== null) {
-        validParams[key] = value;
-      }
-    }
-
-    if (JSON.stringify(validParams) !== JSON.stringify(this.state.params)) {
-      this.setState({ params: validParams }, () => this.setState({ // eslint-disable-line react/no-did-update-set-state
-        queryConfig: this.config() }));
-    }
+    // Remove parameters with null or undefined values
+    const params = omitBy(list.params, isNil);
 
-    if (sortIdx !== this.state.sortIdx) {
-      this.setState({ sortIdx: sortIdx }); // eslint-disable-line react/no-did-update-set-state
+    if (
+      !isEqual(query, prevProps.query) ||
+      !isEqual(params, this.state.params) ||
+      sortIdx !== this.state.sortIdx
+    ) {
+      const queryConfig = this.getQueryConfig({ params }, query);
```

Comment:
So previously the queryConfig state was only updated if that specific prop had changed, but now it is updated under all these other conditions as well. Is that a problem?

Reply:
Not a problem since if any 1 of the conditions is true, a call to `setState` is made anyway. If part of the state didn't actually change, then it doesn't matter. For example, if `sortIdx` changed, but `queryConfig` didn't, the call to `setState` is required anyway. Including a value for `queryConfig` that is the same as the existing state value won't matter.

By combining everything into a single call to `setState` we never have to make multiple calls to it when multiple conditions are true.